storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -139,12 +139,12 @@
                   (.add local pair) 
 
                   ;;Using java objects directly to avoid performance issues in java code
-                  (let [node+port (get @task->node+port task)]
-                    (when (not (.get remoteMap node+port))
-                      (.put remoteMap node+port (ArrayList.)))
-                    (let [remote (.get remoteMap node+port)]
+                  (let []
+                    (when (not (.get remoteMap task))
+                      (.put remoteMap task (ArrayList.)))
+                    (let [remote (.get remoteMap task)]
                       (.add remote (TaskMessage. task (.serialize serializer tuple)))
-                     )))) 
+                    ))))
```

Comment:
> I suggest we do not change L132-137, since it already groups by destination.

I will follow up again, I think I see what you mean in your other comment.